src/solvers/defaults.jl

```diff
@@ -23,9 +21,9 @@ function default_region_printer(;
     region = first(sweep_plan[which_region_update])
     @printf("Sweep %d, region=%s \n", which_sweep, region)
     print("  Truncated using")
-    @printf(" cutoff=%.1E", cutoff)
-    @printf(" maxdim=%d", maxdim)
-    @printf(" mindim=%d", mindim)
+    haskey(inserter_kwargs, :cutoff) && @printf(" cutoff=%.1E", inserter_kwargs.cutoff)
+    haskey(inserter_kwargs, :maxdim) && @printf(" maxdim=%d", inserter_kwargs.maxdim)
+    haskey(inserter_kwargs, :mindim) && @printf(" mindim=%d", inserter_kwargs.mindim)
```

Comment:
This kind of code doesn't feel quite right to me, since it is the same expression repeated multiple times.

Instead, we could turn this into a function and a loop, say:
```julia
print_formatted(x) = print(x)
print_formatted(x::AbstractFloat) = @printf("%.1E", x)
```
and then:
```julia
for key in [:cutoff, :maxdim, :mindim]
  if haskey(inserter_kwargs, key)
    print(key, "=")
    print_formatted(inserter_kwargs[key])
  end
end
```
(I'm not sure if I'm getting the new lines/spacing correct but I think you get the idea.)

Reply:
I don't mind making that change. The only thing about it is that we were making a kind of 'custom' choice for how many digits of the cutoff to print there, just for that one function (the printer, I mean), which is why I was ok with just having it hard coded. But I do agree this is more elegant code and maybe we will find we used `print_formatted` elsewhere like in another printer that we define in that file.

Reply:
(I just mistakenly pushed but didn't make all the changes. Please don't review yet until I do another push & I'll let you know.)

Reply:
Ok now I made all the changes

Reply:
Yes, that's true that what I suggested hardcodes the number of digits, but we can extend it in the future to allow customization, say by passing the number of digits as a keyword argument and using `Val` dispatch to create definitions like: `ndigits(::Val{:cutoff}) = 1`.

Reply:
I like that idea. See also if you like the tweak I made to it, having it return a string instead of directly printing.

Reply:
Looks good to me.